com.civicrm.hrjobroles/tests/phpunit/api/v3/HrJobRolesTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+require_once 'HrJobRolesTestBase.php';
+
+class api_v3_HrJobRolesTest extends HrJobRolesTestBase {
+
+
+  function setUp() {
+    parent::setUp();
+  }
+
+  function tearDown() {
+    parent::tearDown();
+  }
+
+  /**
+   * check with empty array
+   */
+  function testJobRoleCreateEmpty() {
+    $this->callAPIFailure('HrJobRoles', 'create', array());
+  }
+
+  /**
+   * check if required fields are not passed
+   */
+  function testJobRoleCreateWithoutJobContract() {
+    $params = array(
+      'title' => 'test role',
+    );
+    $this->callAPIFailure('HrJobRoles', 'create', $params);
+  }
+
+  /**
+   *  Test creating new role with valid parameters
+   */
+  function testJobRoleCreate() {
+    // create contact
+    $contactParams = array('first_name'=>'walter', 'last_name'=>'white');
+    $contactID = $this->individualCreate($contactParams);
+    // create contract
+    $contract = $this->createJobContract($contactID, date('Y-m-d', strtotime('-14 days')));
+
+    $params = ['job_contract_id' => $contract->id, 'title' => 'test role', 'sequential' => 1];
+    $result = $this->callAPISuccess('HrJobRoles', 'create', $params);
+    $this->assertEquals($result['values'][0]['job_contract_id'], $contract->id);
+    $this->assertEquals($result['values'][0]['title'], 'test role');
+  }
+
+  /**
+   *  Test creating new role with valid
+   * (location, region, department, level, cost center ) fields
+   *
+   */
+  function testJobRoleCreateWithValidOptionValues() {
+    // create contact
+    $contactParams = array('first_name'=>'walter', 'last_name'=>'white');
+    $contactID = $this->individualCreate($contactParams);
+    // create contract
+    $contract = $this->createJobContract($contactID, date('Y-m-d', strtotime('-14 days')));
+
+    // create option group and values
+    $this->createSampleOptionGroupsAndValues();
+
+    $params = [
+      'job_contract_id' => $contract->id,
+      'title' => 'test role',
+      'location' => 'amman',
+      'region' => 'south amman',
+      'department' => 'amman devs',
+      'level_type' => 'guru',
+      'cost_center' => 'abdali',
+      'sequential' => 1
+    ];
+    $result = $this->callAPISuccess('HrJobRoles', 'create', $params);
+    $this->assertEquals($result['values'][0]['location'], 'amman');
+    $this->assertEquals($result['values'][0]['region'], 'south amman');
+    $this->assertEquals($result['values'][0]['department'], 'amman devs');
+    $this->assertEquals($result['values'][0]['level_type'], 'guru');
+    $this->assertEquals($result['values'][0]['cost_center'], 'abdali');
+  }
+
+  /**
+   *  Test creating new role with Invalid
+   * (location, region, department, level, cost center ) fields
+   *
+   */
+  function testJobRoleCreateWithInvalidOptionValues() {
+    // create contact
```

Comment:
The thing is and for example when you try to create new job role from the UI , the code that process the form is using the API to store the job role and not the BAO method . So let assume someone again changed DAO/XML files for the job roles and removed the pusdoconstant from the fields without changing other parts of the code then and if these API tests are removed  all the tests will pass and the system will break due to this change without the developer realize that , but if I kept them then this test specially testJobRoleCreateWithInvalidOptionValues will fail so the developer know that removing the psudoconstant is wrong without changing the other part of the code. so I still think these tests are still important . 

I hope you understood my example above :laughing: